platforms/jvm/scala/src/main/java/org/gradle/api/tasks/scala/ScalaDoc.java

```diff
@@ -207,9 +186,9 @@ public Property<JavaLauncher> getJavaLauncher() {
     @TaskAction
     protected void generate() {
         ScalaDocOptions options = getScalaDocOptions();
-        if (!GUtil.isTrue(options.getDocTitle())) {
-            options.setDocTitle(getTitle());
-        }
+        String docTitle = !GUtil.isTrue(options.getDocTitle().getOrNull())
+            ? getTitle().getOrNull()
+            : options.getDocTitle().getOrNull();
```

Comment:
How about:

```suggestion
    String docTitle = options.getDocTitle()
            .orElse(getTitle())
            .getOrNull();
```

Yes, this would accept `""` as a valid title, but I think that's actually better behavior. That's what we are doing with the footer below anyway.

Reply:
Sure why not. I don't expect this to be a big problem